frontend/src/sharedMethods.js

```diff
@@ -151,17 +151,52 @@ export default {
       return aNumber - bNumber; // Numeric sorting
     });
   },
-  sortSchoolListByTranscriptsAndMincode(schoolsList) {
-    if (!schoolsList) return [];
+  getSchoolOpenStatus(school) {
+    if (!school) return 'Closed'
+
+    const openedDate = new Date(school?.openedDate);
+    const closedDate = school?.closedDate
+      ? new Date(school?.closedDate)
+      : null;
+    const currentDate = new Date();
 
+    if (openedDate <= currentDate && !closedDate) 
+      return 'Open'
+    if (openedDate <= currentDate && closedDate && currentDate < closedDate)
+      return 'Closing'
+    if (currentDate < openedDate)
+      return 'Opening'
+    if (closedDate && closedDate < currentDate)
+      return 'Closed'
+
+    else return 'Closed'
+  },
+  sortSchoolListByCertTranscriptsAndOpenStatusAndMincode(schoolsList) {

```

Comment:
can we rename this to sortSchoolsList() for readability? We don't need to describe how we are sorting in detail in the function name, and the sort parameters might be subject to change so it's best to keep it simple.